music21/musicxml/xmlToM21.py

```diff
@@ -3939,6 +3939,14 @@ def xmlNotationsToSpanners(self, mxNotations, n):
                                     )
             self.setColor(mxObj, slur)
 
+        for mxObj in mxNotations.findall('technical/hammer-on'):
+            self.xmlOneSpanner(mxObj, n, articulations.HammerOn)
+            n.articulations.append(articulations.HammerOn())
+
+        for mxObj in mxNotations.findall('technical/pull-off'):
+            self.xmlOneSpanner(mxObj, n, articulations.PullOff)
+            n.articulations.append(articulations.PullOff())
```

Comment:
Ah. We need to append the same articulation object returned by xmlOneSpanner, not a fresh object.